### DISCUSSION TITLE
Change sprite of OR and AND and XOR to NOR, NAND and XNOR when output is 0

### DISCUSSION THREAD

**Discussion by FalcoGer** [FakeFishGames/Barotrauma#9934] [Ideas and suggestions]
It's sometimes hard to keep track of which logic gates do what. To help with that, the gates could change the text on them if the output is reversed. "OR" would become "NOR" if the output is 0, for example. With a simple change in the sprite you can make logic more clear and remove some of the confusion.

**Comment by mygamingaccount** [comment]
set_output is often used as an additional AND term. This change would lead to a lot of flickering components.

Perhaps if the sprite changed based on whether or not the false output was set to anything nonzero

**Reply by FalcoGer** [reply]
I didn't think of that because I never use set output, being more "traditional" with my logic and programming approaches, but you are right.
But since you can't set the false output, it does make sense to base it on that instead. And since we're on it, might as well have some indication that false output is a "no signal" somehow, too. Because that is not how logic gates work in the real world and it catches me every time I plonk down circuits without thinking about that absurdity.
